src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -277,7 +277,8 @@ private static void AddPopParams(AuthenticationRequestParameters authenticationR
         {
             MsalTokenResponse msalTokenResponse = null;
 
-            if (authResult.IsSuccess)
+            if (authResult.IsSuccess ||  
+                (ResponseStatus)authResult.Error.Status == ResponseStatus.UserSwitch)
```

Comment:
>ResponseStatus)authResult.Error.Status == ResponseStatus.UserSwitch [](http://example.com/codeflow?start=17&length=67)

Does application knows about this userSwitch status? We also need to make sure they are aware of that.

Reply:
No, MSALs consider this as a success case. Altering this behavior would be a breaking change.  

App developers compare `login_hint` with `AuthResult.Username` to find out today.